Automattic/wp-calypso
# Remove donation toggle

Related to #80597

## Proposed Changes

* This diff removes the "Mark this plan as a donation" toggle and related code that uses the toggle's state to potentially update the autogenerated plan name.

## Testing Instructions
Before switching to this branch, you will need to create a payment plan and mark it as a donation.

1. On a site with a sufficient plan, visit Tools > Earn.
2. Click "Manage Payment Button"
3. Click to add a new payment plan, selecting the donation toggle.

Switch to the branch at this point.

1. Go back to the Payment Plans page. The donation plan should have a "Donation" badge on it.
![Screen Shot 2023-08-26 at 3 15 35 PM](https://github.com/Automattic/wp-calypso/assets/8002138/b2c93b08-ff87-4714-9efa-383df7507bec)

2. Click the button to add a new payment plan again. The donation toggle should now be gone from the modal.
![Screen Shot 2023-08-26 at 3 16 09 PM](https://github.com/Automattic/wp-calypso/assets/8002138/c0d3cc8c-7d26-4415-99c7-b6e039a059e2)

## Pre-merge Checklist
